packages/web3-shared/evm/hooks/useBalance.ts

```diff
@@ -1,8 +1,43 @@
 import { useWeb3StateContext } from '../context'
+import { useWeb3 } from '../hooks'
+import { ChainId, ProviderType, NetworkType } from '../types'
+import { useAsync } from 'react-use'
+import { getChainIdFromNetworkType } from '../utils'
+import { getRPCConstants } from '../constants'
+import { first } from 'lodash-es'
 
 /**
  * Get the current block number
  */
 export function useBalance() {
     return useWeb3StateContext().balance
 }
+
+export function useChainBalance(account: string, chainId: ChainId | null, providerType: ProviderType) {
+    const web3 = useWeb3(true, chainId ?? ChainId.Mainnet)
+    return useAsync(async () => (chainId ? web3.eth.getBalance(account) : null), [account, chainId, providerType])
+}
+
+export function useChainBalanceList(account: string, providerType: ProviderType, disabled: boolean) {
```

Comment:
If the caller needs to depress the function, pass an empty account address to replace the `disabled` parameter.